qemu/tests/numa_memdev_options.py

```diff
@@ -111,10 +124,29 @@ def run(test, params, env):
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment
     """
-    error_context.context("Check host's numa node(s) amount!", logging.info)
-    check_host_numa_node_amount(test)
-
+    error_context.context("Check host's numa node(s)!", logging.info)
+    node_list_size_not_zero = check_host_numa_node(test)
+    node1 = node_list_size_not_zero[0]
+    node2 = node_list_size_not_zero[1]
+
+    if params.get('policy_mem') != 'default':
+        error_context.context("Assign host's numa node(s)!", logging.info)
+        params['host-nodes_mem0'] = node1
+        params['host-nodes_mem1'] = node2
+
+    if params.get('set_node_hugepage') == 'yes':
+        h = HugePageConfig(params)
+        hg_pz = str(h.get_hugepage_size())
+        h.set_node_num_huge_pages(num=params['target_num_node0'],
+                                  node=node1, pagesize=hg_pz)
+        if 'numa_hugepage' in params['shortname']:
+            h.set_node_num_huge_pages(num=params['target_num_node1'],
+                                      node=node2, pagesize=hg_pz)
+        h.mount_hugepage_fs()
```

Comment:
Better to clean up the env after test.

Reply:
If keep param "setup_hugepages = yes", hugepages could be set properly in preprocess_vm, then there is no need to set it here. Have tried on my host.